tasks/install-matlab/v1/src/matlab.ts

```diff
@@ -70,18 +110,39 @@ async function resolveLatest(): Promise<string> {
     });
 }
 
-export async function setupBatch(platform: string): Promise<void> {
-    const batchInstallDir = script.defaultInstallRoot(platform, "matlab-batch");
-    const exitCode = await script.downloadAndRunScript(platform, "https://ssd.mathworks.com/supportfiles/ci/matlab-batch/v0/install.sh", batchInstallDir);
+export async function setupBatch(platform: string, architecture: string) {
+    if (architecture !== "x64") {
+        return Promise.reject(Error(`This action is not supported on ${platform} runners using the ${architecture} architecture.`));
+    }
 
-    if (exitCode !== 0) {
-        return Promise.reject(Error(`Failed to install matlab-batch. Script exited with non-zero code ${exitCode}.`));
+    const matlabBatchRootUrl: string = "https://ssd.mathworks.com/supportfiles/ci/matlab-batch/v1/";
+    let matlabBatchUrl: string;
+    let matlabBatchExt: string = "";
+    switch (platform) {
+        case "win32":
+            matlabBatchExt = ".exe";
+            matlabBatchUrl = matlabBatchRootUrl + "win64/matlab-batch.exe";
+            break;
+        case "linux":
+            matlabBatchUrl = matlabBatchRootUrl + "glnxa64/matlab-batch";
+            break;
+        case "darwin":
+            matlabBatchUrl = matlabBatchRootUrl + "maci64/matlab-batch";
+            break;
+        default:
+            return Promise.reject(Error(`This action is not supported on ${platform} runners.`));

```

Comment:
Not sure whether this is user visible. But should the sentence use action, task, argument, ...?